lambdaclass/curse_of_myrra
# Fix SocketConnectionManager Singleton

Fixes some of the issues related to #448

The SocketConnectionManager is intended to be a Singleton object, but until now it was possible to have more than one instance when finishing a game and starting a new one.

Also, a back button was added to the character selection scene.

## Steps to test:
1. Launch a new game in your editor. Check that you have only one instance of the NetworkManager in the character selection scene, that is also alive in the BackendPlayground scene.
2. Pause the game and go back to the lobbies scene. Verify that the NetworkManager was destroyed in this step.
3. Launch a new game, and you should see only one instance of the NetworkManager in the character selection scene.

When you are in the character selection scene, you can also test the back button, that should take you back to the lobbies scene.
